lib/ansible/playbook/conditional.py

```diff
@@ -95,25 +95,24 @@ def evaluate_conditional(self, templar, all_vars):
         False if any of them evaluate as such.
         '''
 
-        # since this is a mix-in, it may not have an underlying datastructure
-        # associated with it, so we pull it out now in case we need it for
-        # error reporting below
+        # since this is a mix-in, it may not have an underlying datastructure associated with it,
+        # so we pull it out now in case we need it for error reporting below
         ds = None
         if hasattr(self, '_ds'):
             ds = getattr(self, '_ds')
 
-        try:
-            # this allows for direct boolean assignments to conditionals "when: False"
-            if isinstance(self.when, bool):
-                return self.when
-
-            for conditional in self.when:
+        # its always a list at this point
+        for conditional in self.when:
+            try:
                 if not self._check_conditional(conditional, templar, all_vars):
                     return False
-        except Exception as e:
-            raise AnsibleError(
-                "The conditional check '%s' failed. The error was: %s" % (to_native(conditional), to_native(e)), obj=ds
-            )
+            except AnsibleError as e:
+                raise AnsibleError(to_native(e), obj=ds)
```

Comment:
Aren't we losing some good "text" here? Shouldn't probably do more than just `to_native(e)`

Reply:
not really, we were just adding redundant information as the errors from the function already explain the issue well enough.